tools/cldr-apps/js/src/views/AddValue.vue

```diff
@@ -0,0 +1,108 @@
+<script setup>
+import { nextTick, ref } from "vue";
+
+import * as cldrAddValue from "../esm/cldrAddValue.mjs";
+
+const xpstrid = ref(""); // xpath string id
+const newValue = ref("");
+const formLeft = ref(0);
+const formTop = ref(0);
+const formIsVisible = ref(false);
+const inputToFocus = ref(null);
+
+function setXpathStringId(id) {
+  xpstrid.value = id;
+}
+
+function showModal(event) {
+  // Use the coordinates of the button's top-left corner
+  formLeft.value = event.clientX - event.offsetX;
+  formTop.value = event.clientY - event.offsetY;
+  newValue.value = "";
+  formIsVisible.value = true;
+  cldrAddValue.setFormIsVisible(true, xpstrid.value);
```

Comment:
"form is visible" would be ideally kept in the presentation (vue) layer rather than owned by the "library".  Couldn't it be a property on this AddValue component? Maybe for a future refactor.

Reply:
yes...

the overall situation is that we have vue components embedded in legacy components, and the latter can get clobbered when a response arrives

setFormIsVisible enables cldrSurvey.isInputBusy to prevent that disruption by checking cldrAddValue.isFormVisible

if we moved that into vue, then cldrSurvey would need to know about vue -- problematic for encapsulation and further mixing of modern/legacy code

long-term solution is more extensive modernization; in the meantime there are compromises